synapse/handlers/oidc.py

```diff
@@ -827,7 +827,7 @@ async def handle_oidc_callback(
             logger.debug("Exchanging OAuth2 code for a token")
             token = await self._exchange_code(code)
         except OidcError as e:
-            logger.exception("Could not exchange OAuth2 code")
```

Comment:
This one can be an indicator of a misconfiguration, or just happen if there is a problem communicating to the IdP? It should not be that common (or we have other things to worry about)

Reply:
We'll still log this as a warning with the proposed change: `Could not exchange OAuth2 code: bad_verification_code: The code passed is incorrect or expired`.